cms/static/cms/js/modules/cms.pagetree.js

```diff
@@ -664,6 +644,74 @@ var CMS = window.CMS || {};
         },
 
         /**
+         * Sets events for the search on the header.
+         *
+         * @method _setSearch
+         * @private
+         */
+        _setSearch: function () {
+            var that = this;
+
+            var filterActive = false;
+            var filterTrigger = this.ui.container.find('.js-cms-pagetree-header-filter-trigger');
+            var filterContainer = this.ui.container.find('.js-cms-pagetree-header-filter-container');
+            var filterClose = filterContainer.find('.js-cms-pagetree-header-search-close');
+            var filterClass = 'cms-pagetree-header-filter-active';
+
+            var visibleForm = this.ui.container.find('.js-cms-pagetree-header-search');
+            var hiddenForm = this.ui.container.find('.js-cms-pagetree-header-search-copy form');
+
+            var searchContainer = this.ui.container.find('.cms-pagetree-header-filter');
+            var searchField = searchContainer.find('#field-searchbar');
+            var timeout = 200;
+
+            // add active class when focusing the search field
+            searchField.on('focus', function (e) {
+                e.stopImmediatePropagation();
+                searchContainer.addClass(filterClass);
+            });
+            searchField.on('blur', function (e) {
+                e.stopImmediatePropagation();
+                // timeout is required to prevent the search field from jumping
+                // between enlarging and shrinking
+                setTimeout(function () {
+                    if (!filterActive) {
+                        searchContainer.removeClass(filterClass);
+                    }
+                }, timeout);
+                that.ui.document.off(that.click);
```

Comment:
would prefer this to be scoped a bit further, purely because `that.click` might be used eventually in other places in the pagetree and it would remove all of the handlers